### ISSUE TITLE
Unable to install dependencies on server 

### ISSUE THREAD

**Issue by mdAliMaaz** [piyushgarg-dev/cloud-ide#5]
![image](https://github.com/piyushgarg-dev/cloud-ide/assets/130007307/2944ca4b-1e6b-4fe3-bc81-78ff0da7870e)

**Comment by kumarkshitij171** [comment]
Problem is with `npm i node-pty`  because it is giving the error.

**Comment by varshil-shah** [comment]
To resolve the issue with the `node-pty` package, ensure that the necessary dependencies are installed on your Ubuntu/Linux system:

```bash
sudo apt update
sudo apt install make
sudo apt install gcc
sudo apt install g++
sudo apt install python3
```

NOTE
> For windows os, I guess we've to install software like [Visual Studio](https://visualstudio.microsoft.com/downloads/) and many other heavy softwares.

**Comment by ruru-m07** [comment]
If you face this issue mean you are a windows user we already discussed on it in issue #4 

Windows user face this issue from many years but still we can't find any solution soo the best why to face this is switch your OS like Ubuntu is best.

**Comment by ruru-m07** [comment]
#4

**Comment by varshil-shah** [comment]
> If you face this issue mean you are a windows user we already discussed on it in issue #4
> 
> Windows user face this issue from many years but still we can't find any solution soo the best why to face this is switch your OS like Ubuntu is best.

Yes, I agree development on windows is headache compare to linux but you can use WSL or use Cloud code editor like replit or else spin up a machine on cloud.

**Comment by rahul-technogetic** [comment]
@ruru-m07  have you tried running the backend code on codespaces? as their are no errors , but from the frontend it is showign the cors issue even after configuring the cors and options for it as per my frontend url!

**Comment by ruru-m07** [comment]
@rahul-technogetic 

> @ruru-m07 have you tried running the backend code on codespaces?

yes, I tried to run code in `github codespaces` and `google IDX` You can check photos on pr #2 

> as there are no errors, but from the front end it is showing the cors issue even after configuring the cors and options for it as per my frontend URL!

can you provide me a little bit more info like a code snippet or some images that help me to understand what's going on? so I can help you. 😊

**Comment by rahul-technogetic** [comment]
> @rahul-technogetic
> 
> > @ruru-m07 have you tried running the backend code on codespaces?
> 
> yes, I tried to run code in `github codespaces` and `google IDX` You can check photos on pr #2
> 
> > as there are no errors, but from the front end it is showing the cors issue even after configuring the cors and options for it as per my frontend URL!
> 
> can you provide me a little bit more info like a code snippet or some images that help me to understand what's going on? so I can help you. 😊

yes here is the Issue that i got in the console 
`Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://animated-palm-tree-976gw6x6pw4wfw44-9000.app.github.dev/socket.io/?EIO=4&transport=polling&t=OzO6eSs. (Reason: CORS request did not succeed). Status code: (null).`

**Comment by ruru-m07** [comment]
@rahul-technogetic it looks like I already faced this issue and I solved this but now your `Status code` is null it should be `301`

it's hard to fix this issue with this information and also descution section is off in this repo.

So can you provide something that help me to direct content with you? like where we can talk about it. 

it's hard to discuss here. if you confutable then content me on IG or another platform

**Comment by ruru-m07** [comment]
## resolved #5

### [https://gist.github.com/ruru-m07/10c8d80ccd77b3eebbc0495165e7b561](https://gist.github.com/ruru-m07/10c8d80ccd77b3eebbc0495165e7b561)

@everyone
@piyushgarg-dev 
@mdAliMaaz 
@kumarkshitij171 
@varshil-shah

**Comment by rajdama** [comment]
@piyushgarg-dev 
@mdAliMaaz
@kumarkshitij171
@varshil-shah 
@ruru-m07 
Here is the solution https://www.linkedin.com/feed/update/urn:li:activity:7216473523446292480/
For now refer to this post, I will shortly add detailed solution properly in here....

**Comment by aayushsingh7** [comment]
I have tried everything to solve this issue but nothing really worked. Then i tried installing and using it in `LINUX` instead of `WINDOWS` and it worked pretty smoothly without any errors

**Comment by rajdama** [comment]
> I have tried everything to solve this issue but nothing really worked. Then i tried installing and using it in `LINUX` instead of `WINDOWS` and it worked pretty smoothly without any errors

You mean wsl?
